c3web-server/src/main/scala/com/ifunsoftware/c3web/routing/ApiRouterActor.scala

```diff
@@ -0,0 +1,52 @@
+package com.ifunsoftware.c3web.routing
+
+/**
+ * Created by alexander on 9/21/2015.
+ */
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props }
+import spray.routing.HttpService
+
+/**
+ * Factory method for Props configuration files for actors
+ */
+object ApiRouterActor {
+  def props(authRoute: ActorRef, accRoute: ActorRef, groupsRoute: ActorRef, groupRoute: ActorRef,
+            filesRoute: ActorRef, fileRoute: ActorRef, downloadRoute: ActorRef,
+            annotationRoute: ActorRef, journalRoute: ActorRef): Props =
+    Props(new ApiRouterActor(authRoute, accRoute, groupsRoute, groupRoute, filesRoute, fileRoute, downloadRoute, annotationRoute, journalRoute))
+}
+
+/**
+ * Routes the incoming request.  If the route begins with "api" the request is passed
+ * along to the matching spray routing actor (if there's a match)
+ */
+class ApiRouterActor(authRoute: ActorRef, accRoute: ActorRef, groupsRoute: ActorRef, groupRoute: ActorRef,
+                     filesRoute: ActorRef, fileRoute: ActorRef, downloadRoute: ActorRef,
+                     annotationRoute: ActorRef, journalRoute: ActorRef) extends Actor
+  with HttpService
+  with ActorLogging {
+
+  def actorRefFactory = context
+  def receive = runRoute {
+    compressResponseIfRequested() {
+      pathPrefix("api") {
+        log.debug("Handling API request")
+        pathPrefix("auth") { ctx => authRoute ! ctx } ~
+          pathPrefix("user") { ctx => accRoute ! ctx } ~
```

Comment:
the naming is a bit misleading: `authRoute`, `accRoute` and other objects are in fact HttpServices, not routes.
I'd suggest to turn them into Routes. Then you don't have to explicitly pass `RoutingContext` `ctx` objects:

```
pathPrefix("bla") {
   authRoute
}
```